### PR TITLE
`Programming exercises`: Fix "Go to Build Plan" URL in the exercise overview

### LINKED ISSUE #5431: Go to build plan takes you to Artemis and not Build Plan
### Describe the bug

When clicking the "Go to build plan" button from the "courses/.../excercises" view, the Button does not take you to the build view, but to "https://artemis.ase.in.tum.de".

If you click the button from the "courses/.../excercises/..." view, the Button does take you to the build view.

It's nothing major, but it gets pretty annoying always having to open the exercise view.
I'll attach some pictures.

### To Reproduce

I'm sorry I don't really know how, I guess you must have such a tab where the build button is visible. I hope that it'll be clear from the Screenshots

### Expected behavior

I was expecting the button to directly take me to the build plan

### Screenshots

**This button takes me to the Artemis homepage:**
From the view where all exercises are listed.
this button takes to artemis

**This button takes me to the Build Plan as expected:**
From the view of this specific exercise.
this button takes to build plan

### What browsers are you seeing the problem on?

Chrome, Firefox

### Additional context

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by Strohgelaender**
#### General
- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.

### Motivation and Context
Closes #5431

### Description
The buildPlanURL was undfined in the reported case. It's a helper attribute and needs to be calculated in the client. I added a call to `setBuildPlanUrlForProgrammingParticipations` to calculate the needed value. 

### Steps for Testing
Replicate the issue. 

Prerequisites:
- 1 Programming Exercise with "Publish Build Plans" enabled

1. Navigate to the exercise overview
2. Start the exercise -> the button should appear and the "Go to Build Plan" link should work
3. Click on the exercise detail -> link should work
4. As instructor, cleanup the participation and resume as student -> link should work
5. When the exercise is shown as a "featured exercise" on the front page, the link should work

#### Code Review
- [ ] Review 1
- [ ] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### CODE REVIEW

**Review by pantageepapa — COMMENTED** [review]
Hi I was not sure about what you meant by "go to build plans", if the links in the screenshot are meant by it then the links work as described :)
Screenshot 2022-07-19 at 21 22 17

**Comment by ole-ve** [comment]
I assume the URL from a student's perspective is broken (and not from an instructor view).

To display the URL, the exercise has to have the option "Publish Build Plan" enabled.

**Review by Hialus — DISMISSED** [review]
Found an issue: 
When you resume an exercise (you can make this button available easily by going to participations and cleanup the participation) and after Artemis recreated the build plan the button has an empty href tag: `<a _ngcontent-bxr-c108="" target="_blank" rel="noreferrer" class="btn btn-primary me-1 ng-star-inserted" style="margin-left: 8px;" href="">`.
Otherwise, I could not find an issue and this is likely not caused by this PR, but maybe you want to fix this as well. If you do this in another PR hit me up and I'll approve this one

**Review by JohannesStoehr — APPROVED** [review]
Changes look good 👍

File: src/main/webapp/app/overview/course-exercises/course-exercise-row.component.ts [hunk 1]
```diff
@@ -51,6 +51,7 @@ export class CourseExerciseRowComponent implements OnInit, OnDestroy, OnChanges
         private exerciseService: ExerciseService,
         private httpClient: HttpClient,
         private participationWebsocketService: ParticipationWebsocketService,
+        private profileService: ProfileService,
```
Comment (JohannesStoehr) [inline src/main/webapp/app/overview/course-exercises/course-exercise-row.component.ts:54]: Why do you need the profileService here?

Comment (Strohgelaender) [reply src/main/webapp/app/overview/course-exercises/course-exercise-row.component.ts:54]: I'm still looking if there is a better place, but right now it seems like I need this service (and the call to `setBuildPlanUrlForProgrammingParticipations` twice). Once when getting participation updates (after pressing start or resume) and once for the initial page load (if the exercise is already started). It seems like these two are completely different paths, but again I'm still looking into this.

File: src/main/webapp/app/exercises/shared/exercise/exercise.service.ts [hunk 2]
```diff
@@ -464,6 +473,15 @@ export class ExerciseService {
         }
     }
 
+    private setBuildPlanUrlToParticipations(exercise: Exercise | undefined | null) {
+        if (exercise?.type === ExerciseType.PROGRAMMING) {
```
Comment (JohannesStoehr) [inline src/main/webapp/app/exercises/shared/exercise/exercise.service.ts:477]: Wouldn't it be enough to do this if the publication of the buildplan is activated for that exercise?

**Review by LiaLism — APPROVED** [review]
The URL worked. Tested on ts1

**Review by JohannesStoehr — APPROVED** [review]
Code looks good to me. Thanks for incorporating my latest suggestion

**Review by b-fein — APPROVED** [review]
Code :+1:

**Pull request merged**